File: backend/Models/ZahtjeviVanredniOtpisDetalji.cs
using Microsoft.EntityFrameworkCore;

namespace backend.Models {
    public class ZahtjeviVanredniOtpisDetalji {
       // public string BrojOtpisa { get; set; } = null!;
        public int ArtikalID { get; set; }
        public string Sifra { get; set; } = null!;
        public string DatumPopunjavanja { get; set; } = null!;

        [Precision(18,2)]
        public decimal Kolicina { get; set; }
        public string PotrebnoZbrinjavanje { get; set; } = null!;
        public string PotrebanTransport { get; set; } = null!;

        public string RazlogOtpisa { get; set; } = null!;

        public string NazivArtikla { get; set; } = null!;
        public string Status { get; set; } = null!;
        public string Dobavljac { get; set; } = null!;

        [Precision(18,2)]
        public decimal NabavnaVrijednost { get; set; }

        [Precision(18,2)]
        public decimal UkupnaVrijednost { get; set; }
    }
}

File: auro/src/app/@core/data/zahtjevi-vanredni-otpis-detalji.ts
export interface ZahtjeviVanredniOtpisDetalji {
  odobreniArtikli: Array<
  {
  artikalId: number,
  sifra: string;
  datumPopunjavanja: string;
  kolicina: number;
  potrebnoZbrinjavanje: string;
  potrebanTransport: string;
  razlogOtpisa: string;
  nazivArtikla: string;
  status: string;
  dobavljac: string;
  nabavnaVrijednost: number;
  ukupnaVrijednost: number;
  }>,

  odbijeniArtikli: [
    {
      odobreniArtikli: Array<{
        sifra: string;
        naziv: string;
        razlogOtpisa: string;
        potrebnoZbrinjavanje: string;
        potrebanTransport: string;
        jedinicaMjere: string;
        kolicina: number;
        nabavnaVrijednost: number;
        ukupnaVrijednost: number;
        dobavljac: string;}>;
        odbio: string;
        komentar: string;
      }
  ]
}
